recipes/libzip/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+        "fPIC": True,
+        "with_bzip2": True,
+        "with_lzma": True,
+        "with_zstd": True,
+        "crypto": "auto"
+    }
+
+    @property
+    def _crypto(self):
+        if self.options.crypto == "auto":
+            return "win32" if self.settings.os == "Windows" else "openssl"
+        return self.options.crypto
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self._crypto == "win32" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("Windows is required to use win32 crypto libraries")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+
+        if self.options.with_bzip2:
+            self.requires("bzip2/1.0.8")
+
+        if self.options.with_lzma:
+            self.requires("xz_utils/5.2.4")
+
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.5 ")
+
+        if self._crypto == "openssl":
+            self.requires("openssl/1.1.1g")
+        elif self._crypto == "mbedtls":
+            self.requires("mbedtls/2.16.3-gpl")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_REGRESS"] = False
+        cmake.definitions["BUILD_EXAMPLES"] = False
+        cmake.definitions["BUILD_DOC"] = False
+
+        cmake.definitions["ENABLE_LZMA"] = self.options.with_lzma
+        cmake.definitions["ENABLE_BZIP2"] = self.options.with_bzip2
+        cmake.definitions["ENABLE_ZSTD"] = self.options.with_zstd
+
+        cmake.definitions["ENABLE_COMMONCRYPTO"] = False  # TODO: We need CommonCrypto package
+        cmake.definitions["ENABLE_GNUTLS"] = False  # TODO: We need GnuTLS package
+
+        cmake.definitions["ENABLE_MBEDTLS"] = self._crypto == "mbedtls"
+        cmake.definitions["ENABLE_OPENSSL"] = self._crypto == "openssl"
+        cmake.definitions["ENABLE_WINDOWS_CRYPTO"] = self._crypto == "win32"
+
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_id(self):
+        self.info.options.crypto = self._crypto
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Windows" and self_.crypto == "win32":
+            self.cpp_info.libs.append("bcrypt")
+
```

Comment:
There are some executables in the `bin` subdirectory.
```suggestion
    bin_path = os.path.join(self.package_folder, "bin")
    self.output.info("Appending PATH environment variable: {}".format(bin_path))
    self.env_info.PATH.append(bin_path)
```

Reply:
@Fefer-Ivan 
Can you take a look at this?

Reply:
I tried to add your suggested code and run it locally and it failed like this: 
```
ERROR: libzip/1.7.3: Error in package() method, line 103
	self.env_info.PATH.append(bin_path)
	AttributeError: 'NoneType' object has no attribute 'PATH'
```

Could you please clarify what should package do with binaries? Add them to PATH environment variable?

Thanks.

Reply:
Fixed it. I put the code to package(self) method, instead of package_info(self). It is now pushed to branch.

Reply:
Yup, package_info has cpp_info, env_info and user_info. All _info's are defined there.